src/zenml/integrations/mlflow/model_registries/mlflow_model_registry.py

```diff
@@ -0,0 +1,796 @@
+            description (str, optional): The description of the model.
+            tags (Dict, optional): The tags of the model.
+            model_source_uri (str, optional): The source URI of the model.
+            version (str, optional): The version of the model.
+            version_description (str, optional): The description of the model
+                version.
+            version_tags (Dict, optional): The tags of the model version.
+            registry_metadata (Dict, optional): The registry metadata of the
+                model version.
+            zenml_version (str, optional): The ZenML version.
+            zenml_pipeline_run_id (str, optional): The ZenML pipeline run ID.
+            zenml_pipeline_name (str, optional): The ZenML pipeline name.
+            zenml_step_name (str, optional): The ZenML step name.
+            **kwargs: Additional keyword arguments.
+
+        Raises:
+            MlflowException: If the registered model does not exist.
+
+        Returns:
+            The registered model version.
+        """
+        # Check if the model exists, if not create it.
+        if not self.check_model_exists(name):
+            logger.warning(
+                f"No registered model with name {name} found. Creating a new"
+                "registered model."
+            )
+            self.register_model(
+                name=name,
+                description=description,
+                tags=tags,
+            )
+        try:
+            # Inform the user that the version is ignored.
+            if version:
+                logger.info(
+                    f"MLflow model registry does not take a version as an argument. "
+                    f"Registering a new version for the model `'{name}'` "
+                    f"a version will be assigned automatically."
+                )
+            # Set the tags.
+            if not version_tags:
+                version_tags = {}
+            version_tags["zenml_version"] = zenml_version or __version__
+            version_tags["zenml_pipeline_run_id"] = zenml_pipeline_run_id or ""
+            version_tags["zenml_pipeline_name"] = zenml_pipeline_name or ""
+            version_tags["zenml_step_name"] = zenml_step_name or ""
+            # Register the model version.
+            registered_model_version = self.mlflow_client.create_model_version(
+                name=name,
+                source=model_source_uri,
+                run_id=registry_metadata.get("mlflow_run_id")
+                if registry_metadata
+                else "",
+                run_link=registry_metadata.get("mlflow_run_link")
+                if registry_metadata
+                else "",
+                description=version_description,
+                tags=version_tags,
+            )
+        except MlflowException as e:
+            logger.error(
+                f"Failed to register new model version for model '{name}'."
+            )
+            raise e
```

Comment:
I see many of these blocks where you pass the MLflow error back to the user land, but I think you should wrap them in something that is a bit more generic, like an exception that the user should expect, `RuntimeError` maybe or even a `ModelRegistryError` exception that you define as part of the abstraction. 

The reason for this is that the user doesn't necessarily know that they are using MLflow as the particular model registry implementation and so they won't expect an `MlflowException` to be thrown and won't attempt to catch it. But if you had a more generic exception that they can expect, they can catch that instead.